node/actors/bft/src/testonly/twins/tests.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    collections::{BTreeSet, HashSet},
+    fmt::Debug,
+};
+
+use rand::Rng;
+use zksync_concurrency::ctx;
+
+use super::{partitions, Partitioning};
+
+#[test]
+fn test_partitions() {
+    let got = partitions(&[&1, &2, &3], 2);
+    let want: HashSet<Partitioning<_>> = HashSet::from_iter(vec![
```

Comment:
I kept the inner part as `Vec<Vec<_>>` because:
* I thought it would be an easier syntax to define all the expected cases in the test with `vec![...]`
* the algorithm actually preserves the order to some extent, e.g. the leftmost item is always in the first partition

But I didn't have any particular expectation about the order of in which they appear, so I thought an external `HashSet` can be used to check equivalence from that point.

Maybe I'll add some different labels to remove any doubt that it's from the ordering of the item type.

Reply:
fyi HashSet can be constructed just like `[a,b,c].into()` assuming that rust can deduce the target type.